File: backend/src/main/kotlin.nikolay/otdo/posts/model/Post.kt
package otdo.posts.model

data class Post(
    val id: Long? = null,
    val heading: String,
    val text: String,
    val textColor: String,
    val backgroundColor: String,
    val imgSrc: String,
    val postStyle: PostStyle,
    val postType: PostType,
)

enum class PostStyle {
    PICTURE_LEFT, PICTURE_RIGHT, PICTURE_BOTTOM
}

enum class PostType {
    OVERVIEW, MACHINES, SERVICES, GALLERY
}

File: frontend/src/app/pages/main/model/Post.ts
export interface Post {
  heading: string;
  text: string;
  textColor: string;
  imgSrc: string;
  backgroundColor: string;
  postStyle: PostStyle;
}

export enum PostStyle {
  PICTURE_LEFT = "PICTURE_LEFT",
  PICTURE_RIGHT = "PICTURE_RIGHT",
  PICTURE_BOTTOM = "PICTURE_BOTTOM"
}
